### ISSUE TITLE
Why are video transcodings a lot more inefficient than say Youtube?

### ISSUE THREAD

**Issue by tio-trom** [Chocobozzz/PeerTube#3856]
We made a live yesterday on Peertube. Almost 2 and a half hours long. This is it https://videos.trom.tf/videos/watch/08976e30-bcb8-4660-9322-57a01272a943 - the video 1080p is 5.8 GB in size after conversion, which seems immense to me. We convert in 730p and 1080p. With the same OBS Studio setup, streaming on Youtube, we normally got the same length of videos at 5 times less the size, at about 1GB. We have done over 60 such streams over the past years so I have a great sample size to compare with. Why is that?

I am currently uploading this video on Youtube and will wait for it to convert to see how it does it.

**Comment by TROMsite** [comment]
I am providing you with 2 samples if it helps. Same exact video.

First video is the original Peertube video, converted by peertube - https://www.drive.tromsite.com/s/3pZRjL86dkdqk2o - it is 1080p and 5.4GB in size.

Second video is from Youtube. I uploaded the above Peertube video on youtube and let youtube convert it again. Then I grabbed it from there. The file is 1080p and 1.4GB in size https://www.drive.tromsite.com/s/4jN5aB2jcL6Jiq6

The difference is enormous and the quality seems to be very similar. It would help a ton if the Peertube conversion would be as good. We have over 60 such TROMcasts (videos). On youtube they are around 90GB in size, in total. On Peertube that would be 5 times higher at around 450GB. That's insanely enormous.

**Comment by bkil** [comment]
I would rephrase your question as why the default bitrate (quality) of PeerTube is higher than that of YouTube and whether it would be possible to have a per-account setting for this.

**Comment by TROMsite** [comment]
So it is the bitrate that makes the difference?

**Comment by bkil** [comment]
Also related: #2442, #1899, #1418

**Comment by Chocobozzz** [comment]
> So it is the bitrate that makes the difference?

Yes, we target 5Mb/s bitrate for 1080p videos (https://github.com/Chocobozzz/PeerTube/blob/develop/shared/models/videos/video-resolution.enum.ts#L53). This is the recommended bitrate by youtube :thinking: 

In your specific case, youtube uses a 1.2Mb/s bitrate, it's the reason why the file is lower in size.

As @bkil said, we developed a way for plugins to create their own transcoding profile (in 3.1), so maybe you could try that to reduce the video file size.

**Comment by TROMsite** [comment]
Thanks. I see, so basically if anyone will make a plugin about this, then that's kinda the solution? Unfortunately I am not skilled to help with this

**Comment by TROMsite** [comment]
Let me post a follow-up on this. We had another live streaming. At around one hour and 40 minutes. The end result was this https://www.drive.tromsite.com/s/462Ss73CxcRdS7y 4.1GB in size. Our of curiosity I reuploaded the video on our Peertube instance and Peertube transcoded it again and ended up with a much smaller file size at 1.6GB https://www.drive.tromsite.com/s/cRWRPo7kt6oTCYt. 

The quality seems to be the same, I can't find any differences. But there is a huge difference in size. How come?

**Comment by Chocobozzz** [comment]
I think it's because of https://github.com/Chocobozzz/PeerTube/blob/develop/server/lib/video-transcoding-profiles.ts#L46

It uses a CBR set to this value, even if not needed. Maybe we should just use a VBR using the `maxrate` like we do for VOD, to cap the bitrate to a max value, so ffmpeg can easily transcodes the video to a lower value.

Feedbacks welcomed here :)

**Comment by Avatat** [comment]
We should use CBR for live streaming and VBR for saving it on rest, for later watching.

**Comment by TROMsite** [comment]
Thanks for the explanation. I definitely think that this is very important since we are talking about self-hosting and disk space is one of the main concerns when it comes to a self-hosted video platform. If the size of livestreams can be reduced by  a factor of 4-5, then that's a huge improvement.

**Comment by Chocobozzz** [comment]
We still use CBR but now cap the bitrate to the input bitrate in 3.4. We also changed the bitrate calculation in https://github.com/Chocobozzz/PeerTube/commit/679c12e69c9f3a2d003ee3abe8b8da49f25b2bd3 (see https://github.com/Chocobozzz/PeerTube/blob/develop/shared/core-utils/videos/bitrate.ts)
